src/pepr/operator/controllers/istio/virtual-service.ts

```diff
@@ -0,0 +1,91 @@
+import { K8s, Log } from "pepr";
+
+import { UDSConfig } from "../../../config";
+import { Gateway, Istio, UDSPackage, getOwnerRef } from "../../crd";
+
+/**
+ * Creates a VirtualService for each exposed service in the package
+ *
+ * @param pkg
+ * @param namespace
+ */
+export async function virtualService(pkg: UDSPackage, namespace: string) {
```

Comment:
I prefer implicit in TS and leveraging the language features (same reason we dont do [Hungarian Notation](https://en.wikipedia.org/wiki/Hungarian_notation) or typed-jsdoc these days in TS-land. Not sure I'm tracking on lack of VSCode hover feature...

![Screenshot 2024-01-09 at 1 28 11 PM](https://github.com/defenseunicorns/uds-core/assets/882485/7222739e-9589-4a42-9b53-3ead0c4a8f68)
![Screenshot 2024-01-09 at 1 28 04 PM](https://github.com/defenseunicorns/uds-core/assets/882485/f27bf780-4bc7-4949-a752-2ace9b0f38e8)

Reply:
As shown in your screenshot, when you hover in VS Code you get to see the type annotations, in a GitHub review you don't get such niceties, so it can be _slightly_ harder for someone to review.

Reply:
"makes review easier" -- ah you were saying GH Review lacks the VSCode feature... 😆

Reply:
teach someone to fish, something, something....

![Screenshot 2024-01-09 at 7 32 38 PM](https://github.com/defenseunicorns/uds-core/assets/882485/e3b922f4-d186-4e78-ba61-c602d203945a)

Reply:
brb, completely shifting my review flow